### PR TITLE
zammad: 5.1.1 -> 5.4.1

### PR DESCRIPTION
**Pull request by n0emis**
###### Description of changes

https://zammad.com/en/releases/5-4-1

ZHF #230712

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by n0emis** [comment]
It does work, but upgrading in place might need some work (e.g. I needed to clean /var/lib/zammad before I got it working).

**Comment by n0emis** [comment]
related to #230868, as zammad now uses ruby 3.1

**Comment by mweinelt** [comment]
And #210452.

File: pkgs/applications/networking/misc/zammad/update.sh [hunk 1]
```diff
@@ -8,7 +8,8 @@ if [ "$#" -gt 2 ] || [[ "$1" == -* ]]; then
   exit 1
 fi
 
-VERSION=$(curl -s https://ftp.zammad.com/ | grep -v latest | grep tar.gz | sed "s/<a href=\".*\">zammad-//" | sort -h | tail -n 1 | awk '{print $1}' | sed 's/.tar.gz<\/a>//')
+#VERSION=$(curl -s https://ftp.zammad.com/ | grep -v latest | grep tar.gz | sed "s/<a href=\".*\">zammad-//" | sort -h | tail -n 1 | awk '{print $1}' | sed 's/.tar.gz<\/a>//')
```
Comment (mweinelt) [inline pkgs/applications/networking/misc/zammad/update.sh:11]: VERSIO=$(xidel -s https://ftp.zammad.com/ --extract "//a" | grep -E "zammad-[0-9.]*.tar.gz" | sort --version-sort | tail -n 1)

Comment (n0emis) [reply pkgs/applications/networking/misc/zammad/update.sh:11]: oops, did'nt mean to commit this :P

Comment (mweinelt) [reply pkgs/applications/networking/misc/zammad/update.sh:11]: Ah, thought you were looking for an alternative to what you have :D

Comment (n0emis) [reply pkgs/applications/networking/misc/zammad/update.sh]: that too...

File: pkgs/applications/networking/misc/zammad/update.sh [hunk 2]
```diff
@@ -8,7 +8,7 @@ if [ "$#" -gt 2 ] || [[ "$1" == -* ]]; then
   exit 1
 fi
 
-VERSION=$(curl -s https://ftp.zammad.com/ | grep -v latest | grep tar.gz | sed "s/<a href=\".*\">zammad-//" | sort -h | tail -n 1 | awk '{print $1}' | sed 's/.tar.gz<\/a>//')
+VERSION=$(xidel -s https://ftp.zammad.com/ --extract "//a" | grep -E "zammad-[0-9.]*.tar.gz" | sort --version-sort | tail -n 1)
```
Comment (mweinelt) [inline pkgs/applications/networking/misc/zammad/update.sh:11]: > zammad-5.4.1.tar.gz

that's not yet the version :)

Comment (n0emis) [reply pkgs/applications/networking/misc/zammad/update.sh]: reminder to self: don't commit if your tired

File: pkgs/applications/networking/misc/zammad/default.nix [hunk 3]
```diff
@@ -129,6 +136,7 @@ stdenv.mkDerivation {
   passthru = {
     inherit rubyEnv yarnEnv;
     updateScript = [ "${callPackage ./update.nix {}}/bin/update.sh" pname (toString ./.) ];
+    passthru.tests.zammad = nixosTests.zammad;
```
Comment (mweinelt) [inline pkgs/applications/networking/misc/zammad/default.nix:139]:
```suggestion
		tests = { inherit (nixosTests) zammad; };
```

passthru.passthru....

**Comment by mweinelt** [comment]
Tests also work on aarch64-linux.

**Comment by RaitoBezarius** [comment]
> It does work, but upgrading in place might need some work (e.g. I needed to clean /var/lib/zammad before I got it working).

Could you check the upgrade instructions? Can we lock the state?

**Review by mweinelt — APPROVED** [review]
lgtm.

**Pull request merged**